### PR TITLE
feat: new brand multiprocesses feature

### PR DESCRIPTION
**Pull request by Desu-potato**
fuck my ass

### CODE REVIEW

**Comment by kbakdev** [comment]
Resolve conflicts

File: .github/workflows/python-app.yml [hunk 1]
```diff
@@ -31,6 +31,3 @@ jobs:
         flake8 . --count --select=E9,F63,F7,F82 --show-source --statistics
         # exit-zero treats all errors as warnings. The GitHub editor is 127 chars wide
         flake8 . --count --exit-zero --max-complexity=10 --max-line-length=127 --statistics
-    - name: Test with pytest
-      run: |
-        pytest
```
Comment (kbakdev) [inline .github/workflows/python-app.yml:36]: A very creative solution to a problem

File: main.py [hunk 2]
```diff
@@ -1,54 +1,57 @@
...
-from rich.console import Console
-from rich.table import Table
 
-import platform
-import sys
-import psutil
 
-import string, time
+#Classes
+class findings:
```
Comment (kbakdev) [inline main.py:11]: I think there are too few public methods

File: main.py [hunk 3]
```diff
@@ -1,54 +1,57 @@
...
 def indexToCharacter(index):
     if NUMBER_OF_CHARACTERS <= index:
         raise ValueError("Index out of range.")
-    
+
     return ALLOWED_CHARACTERS[index]
 
-# The function takes the next list from the interval
-def n(s_list):
+def next(string):
```
Comment (kbakdev) [inline main.py:46]: We are redefining built-in `next` func

Comment (kbakdev) [reply main.py:46]: Otherwise, we're redefining name `string` from outer scope.

File: main.py [hunk 4]
```diff
@@ -60,81 +63,176 @@ def parseMe(array):
             break
```
Comment (kbakdev) [inline main.py:63]: No exception type specified

File: main.py [hunk 5]
```diff
@@ -60,81 +63,176 @@ def parseMe(array):
...
     return "".join(buf)
 
-
+def findMe(jakisint):
```
Comment (kbakdev) [inline main.py:66]: Too many branches

File: main.py [hunk 6]
```diff
@@ -60,81 +63,176 @@ def parseMe(array):
...
+    test = round((len(str(jakisint))/2)-0.5)
+    stars = test
+    map = {}
```
Comment (kbakdev) [inline main.py:69]: Redefining built-in `map`

File: main.py [hunk 7]
```diff
@@ -60,81 +63,176 @@ def parseMe(array):
...
+        buffor.append(indexToCharacter(int(workString)))
+
+    if jakisint == cycles[len(cycles)-1]:
+        buffor = []
+        for _ in range(stars+1):
+            buffor.append("0")
+
+    return buffor
+
+def sortMe(list):
```
Comment (kbakdev) [inline main.py:129]: Redefining built-in `list`

File: main.py [hunk 8]
```diff
@@ -60,81 +63,176 @@ def parseMe(array):
...
+    bufor = []
+    i = len(list)
+    for _ in range(i):
+        bufor.append(0)
+    for rec in list:
+        bufor[rec.i-1] = rec
+    return bufor
 
 def codeMe(data, key):
```
Comment (kbakdev) [inline main.py:138]: Redefining name `data` and `key` for outer scope

File: main.py [hunk 9]
```diff
@@ -60,81 +63,176 @@ def parseMe(array):
...
     cipher = DES.new(key=pad(key, 8), mode=DES.MODE_ECB)
     ciphertext = cipher.encrypt(pad(data, 8))
     return ciphertext
 
-def decodeMe(data, key):
+def deCodeMe(data, key):
```
Comment (kbakdev) [inline main.py:143]: Redefining name `data` and `key` for outer scope

File: main.py [hunk 10]
```diff
@@ -60,81 +63,176 @@ def parseMe(array):
...
-        #         break
-
-
-if __name__ == "__main__":
-    start_time = time.time()
-    main(DATA, KEY)
\ No newline at end of file
+        textbyte = bytearray(parseMe(deCodeMe(pp["Chipher"], key)), "utf-8")
+        if codeMe(textbyte, key) == pp["Chipher"]:
+            record = findings()
```
Comment (kbakdev) [inline main.py:158]: Redefining name `record` from outer scope

File: main.py [hunk 11]
```diff
@@ -60,81 +63,176 @@ def parseMe(array):
...
+            break
+
+    fr = frame()
+    fr.name = (str(pp["StandardMin"]) +" : " + str(pp["StandardMax"]))
+    fr.objects = omega[:]
+    fr.lenMe()
+    fr.i = round(pp["StandardMax"]/100000)
+    return fr
+
+def prepere(number, chipher):
```
Comment (kbakdev) [inline main.py:176]: Redefining `chiper` from outer scope

File: main.py [hunk 12]
```diff
@@ -60,81 +63,176 @@ def parseMe(array):
...
+        data = sys.argv[1]
+        key = sys.argv[2]
+        numberOfCycles = int(sys.argv[3])
+
+    coreNumber = multiprocessing.cpu_count()
+
+
+    chipher = codeMe(bytearray(data, "utf-8"), bytearray(key, "utf-8"))
+    tab = prepere(numberOfCycles, chipher)
+    p = multiprocessing.Pool(coreNumber)
```
Comment (kbakdev) [inline main.py:212]: Consider using `with` for resource-allocating operations.

**Comment by kbakdev** [comment]
Check this out https://realpython.com/documenting-python-code/

File: main.py (hunk 2 above)
Comment (kbakdev) [reply main.py:11]: btw, class name `findings` doesn't conform to PascalCase naming style

File: main.py [hunk 13]
```diff
@@ -1,54 +1,57 @@
...
-import string, time
+#Classes
+class findings:
+    def __init__(self):
+        self.password = 0
+        self.key = 0
+        self.i = 0
+        self.time = 0
+
+class frame:
```
Comment (kbakdev) [inline main.py:18]: Class name `frame` doesn't conform to PascalCase naming style

File: main.py [hunk 14]
```diff
@@ -1,54 +1,57 @@
...
+    def __init__(self):
+        self.name = ""
+        self.i = 0
+        self.len = 0
+        self.objects = []
+
+    def lenMe(self):
```
Comment (kbakdev) [inline main.py:25]: Method name `lenMe` doesn't conform to snake_case naming style

File: main.py [hunk 15]
```diff
@@ -1,54 +1,57 @@
...
+        self.len = len(self.objects)
+
+    def boutMe(self):
```
Comment (kbakdev) [inline main.py:28]: Method name `boutMe` doesn't conform to snake_case naming style

File: main.py [hunk 16]
```diff
@@ -1,54 +1,57 @@
...
-        1253656678 => '1.17GB'
-    """
-    factor = 1024
-    for unit in ["", "K", "M", "G", "T", "P"]:
-        if bytes < factor:
-            return f"{bytes:.2f}{unit}{suffix}"
-        bytes /= factor
 
+#Helper functions
 def characterToIndex(char):
```
Comment (kbakdev) [inline main.py:36]: Function name `characterToIndex` doesn't conform to snake_case naming style

File: main.py [hunk 17]
```diff
@@ -1,54 +1,57 @@
...
     return ALLOWED_CHARACTERS.index(char)
 
+
 def indexToCharacter(index):
```
Comment (kbakdev) [inline main.py:40]: Function name `indexToCharacter` doesn't conform to snake_case naming style

File: main.py [hunk 18]
```diff
@@ -1,54 +1,57 @@
...
-        if characterToIndex(s_list[0]) == 0:
-            return list(s_list[0]) + n(s_list[1:])
-    return s_list
-
+        string[0] = indexToCharacter((characterToIndex(string[0]) + 1) % NUMBER_OF_CHARACTERS)
+        if characterToIndex(string[0]) == 0:
+            return list(string[0]) + next(string[1:])
+    return string
 
 def parseMe(array):
```
Comment (kbakdev) [inline main.py:56]: Function name `parseMe` doesn't conform to snake_case naming style

File: main.py (hunk 5 above)
Comment (kbakdev) [reply main.py:66]: Function name `findMe` doesn't conform to snake_case naming style

File: main.py [hunk 19]
```diff
@@ -60,81 +63,176 @@ def parseMe(array):
...
+    stars = test
+    map = {}
+    map[0] = 0
+    for i in range(test):
+        map[i+1] =  (100**(i+1))
+    cycles = {}
+    buffor = {}
+    for i in range(len(cycles)+1):
+        buffor[i] = 0
+    for i in map:
```
Comment (kbakdev) [inline main.py:77]: Consider iterating with `.items()`

File: main.py [hunk 20]
```diff
@@ -60,81 +63,176 @@ def parseMe(array):
...
+        cycles[i] = map[i]
+    for i in cycles:
+        if i != 0:
+            cycles[i] = cycles[i] + cycles[i - 1]
+    beta = len(cycles)
+    for _ in range(len(cycles)):
+        beta = beta-1
+        buffor[(beta+1)] = cycles[beta]
+    cycles = {}
+    for i in range(len(buffor)):
```
Comment (kbakdev) [inline main.py:87]: Consider using enumerate instead of iterating with range and len

File: main.py [hunk 21]
```diff
@@ -60,81 +63,176 @@ def parseMe(array):
...
+    used = 0
+    if jakisint - cycles[i - 1] < 0:
+        test = jakisint - cycles[i - 2]
+        used = i-2
+
+    if jakisint - cycles[i - 1] > 0:
+        test = jakisint - cycles[i - 1]
+        used = i - 1
+
+    workString = str(test)
```
Comment (kbakdev) [inline main.py:102]: Variable name `workString` doesn't conform to snake_case naming style

File: main.py [hunk 22]
```diff
@@ -60,81 +63,176 @@ def parseMe(array):
...
+    if len(workString) % 2 == 1:
+        workString = "0"+workString
```
Comment (kbakdev) [inline main.py:104]: Variable name `workString` doesn't conform to snake_case naming style

File: main.py [hunk 23]
```diff
@@ -60,81 +63,176 @@ def parseMe(array):
...
+
+    # if stars
+    if used != round(len(workString)/2):
+        i =   used - round(len(workString)/2)
+        workString = ("00"*(i))+workString
```
Comment (kbakdev) [inline main.py:109]: Variable name `workString` doesn't conform to snake_case naming style

File: main.py (hunk 7 above)
Comment (kbakdev) [reply main.py:129]: Function name `sortMe` doesn't conform to snake_case naming style

File: main.py (hunk 8 above)
Comment (kbakdev) [reply main.py:138]: Function name `codeMe` doesn't conform to snake_case naming style

File: main.py (hunk 9 above)
Comment (kbakdev) [reply main.py:143]: Function name `deCodeMe` doesn't conform to snake_case naming style

File: main.py [hunk 24]
```diff
@@ -60,81 +63,176 @@ def parseMe(array):
...
     return ciphertext
 
-def main(data, key):
-    sequence = []
-    cipher = codeMe(bytearray(data,"utf-8"),bytearray(key,"utf-8"))
-    table = Table(title="DISTRIBUTED CIPHER CRACKING")
-    table.add_column("SECONDS", justify="right", style="cyan", no_wrap=True)
-    table.add_column("POSSIBLE PASSWORD", style="magenta")
-    table.add_column("KEY", justify="right", style="green")
+def begin(pp):
```
Comment (kbakdev) [inline main.py:148]: Maybe another name of var for `pp`?

File: main.py [hunk 25]
```diff
@@ -60,81 +63,176 @@ def parseMe(array):
...
+            record.password = textbyte.decode("utf-8")
+            record.key = key.decode("utf-8")
+            record.time = str(time.time() - pp["TimeStamp"])
+            omega.append(record)
+        i = i +1
+        sequence = next(sequence)
+        if i == pp["StandardMax"]:
+            break
+
+    fr = frame()
```
Comment (kbakdev) [inline main.py:169]: Maybe another name of var for `fr`?

File: main.py [hunk 26]
```diff
@@ -60,81 +63,176 @@ def parseMe(array):
...
+    fr.name = (str(pp["StandardMin"]) +" : " + str(pp["StandardMax"]))
+    fr.objects = omega[:]
+    fr.lenMe()
+    fr.i = round(pp["StandardMax"]/100000)
+    return fr
+
+def prepere(number, chipher):
+    buf = []
+    ts = time.time()
```
Comment (kbakdev) [inline main.py:178]: Maybe another name of var for `ts`?

File: main.py [hunk 27]
```diff
@@ -60,81 +63,176 @@ def parseMe(array):
...
+    for i in range(number):
+        bar = {
```
Comment (kbakdev) [inline main.py:180]: I think `bar` is disallowed name

File: main.py [hunk 28]
```diff
@@ -60,81 +63,176 @@ def parseMe(array):
...
+        buf.append(bar)
+    return buf
+
+
+
+
+if __name__ == '__main__':
+    startTimeProgram = time.time()
+    #Base info
+    data = "0"
```
Comment (kbakdev) [inline main.py:198]: Constant name `data` doesn't conform to UPPER_CASE naming style

File: main.py [hunk 29]
```diff
@@ -60,81 +63,176 @@ def parseMe(array):
...
+    key = "0"
```
Comment (kbakdev) [inline main.py:199]: Constant name `key` doesn't conform to UPPER_CASE naming style

File: main.py [hunk 30]
```diff
@@ -60,81 +63,176 @@ def parseMe(array):
...
+    numberOfCycles = 1
```
Comment (kbakdev) [inline main.py:200]: Constant name `numberOfCycles` doesn't conform to UPPER_CASE naming style

File: main.py [hunk 31]
```diff
@@ -60,81 +63,176 @@ def parseMe(array):
...
+
+
+    chipher = codeMe(bytearray(data, "utf-8"), bytearray(key, "utf-8"))
+    tab = prepere(numberOfCycles, chipher)
+    p = multiprocessing.Pool(coreNumber)
+
+    outputList = p.map(begin, tab)
+    '''
+    output Structure:
+
```
Comment (kbakdev) [inline main.py:217]: Trailing whitespace

File: main.py [hunk 32]
```diff
@@ -60,81 +63,176 @@ def parseMe(array):
...
+    outputList = p.map(begin, tab)
+    '''
+    output Structure:
+    
+    outputList list -> frame class obj
+    frame struct:
+        name string #The interval in which the process operated.
+        len #Number of results
+        objects list -> findings class obj #Stores the results of an interval
+
```
Comment (kbakdev) [inline main.py:223]: Trailing whitespace

File: main.py [hunk 33]
```diff
@@ -60,81 +63,176 @@ def parseMe(array):
...
+    '''
+
+    sortlist = sortMe(outputList)
+    for record in sortlist:
+        if record.len != 0:
+            record.boutMe()
+
+
+    print("Program output time: ", str(time.time() - startTimeProgram))
+
```
Comment (kbakdev) [inline main.py:238]: Trailing newlines

**Review by kbakdev — CHANGES_REQUESTED** [review]

File: main.py (hunk 2 above)
Comment (Desu-potato) [reply main.py:11]: naming style to fix, i think we are good with methods

File: main.py (hunk 3 above)
Comment (Desu-potato) [reply main.py:46]: eeeee k

File: main.py (hunk 5 above)
Comment (Desu-potato) [reply main.py:66]: to fix

File: main.py (hunk 31 above)
Comment (Desu-potato) [reply main.py:217]: what?

**Pull request merged**